lib/elixir/lib/kernel/utils.ex

```diff
@@ -65,14 +69,8 @@ defmodule Kernel.Utils do
   @doc """
   Callback for defstruct.
   """
-  def defstruct(module, fields) do
-    case fields do
-      fs when is_list(fs) ->
-        :ok
-      other ->
-        raise ArgumentError, "struct fields definition must be list, got: #{inspect other}"
-    end
-
+  @spec defstruct(module, Keyword.t) :: {struct, keys :: [atom], derive :: [module]}
```

Comment:
No need to spec this function. This is a callback for internal implementation details.